src/roswire/ros2/launch.py

```diff
@@ -167,17 +174,20 @@ def launch(self,
             If the given launch file could not be found in the package.
         """
         shell = self._app_instance.shell
+        filename_without_path = os.path.basename(filename)
         if not args:
             args = {}
         if not launch_prefixes:
             launch_prefixes = {}
-        filename = self.locate(filename, package=package)
 
         if node_to_remappings or launch_prefixes:
             m = "Requires self.read: not yet implemented"
             raise NotImplementedError(m)
-
-        cmd = ['ros2 launch', shlex.quote(filename)]
+        if package:
+            cmd = ['ros2 launch', (package + ' ' + filename_without_path)]
```

Comment:
Use `shlex.quote` to protect against malformed inputs. Also, we join the string using `' '` later, so there's no need for `(foo + ' ' + bar)`. Prefer:

`['ros2 launch', shlex.quote(package), shlex.quote(filename_without_path)]`

Reply:
`filename_without_path` should be moved to inside this `if` block.